src/app/components/nav-bar/nav-bar.component.ts

```diff
@@ -10,17 +10,16 @@ import { Router } from '@angular/router';
 })
 export class NavBarComponent implements OnInit {
 
-  icons = {
-    leftArrow: faChevronLeft
-  };
+  iconS: IconService;
 
   hideBackButton = false;
 
-  constructor(private location: Location, private router: Router) { }
+  constructor(private location: Location, private router: Router, private iconService: IconService) { }
 
   @Input() title = 'default title';
 
   ngOnInit(): void {
+    this.iconS = this.iconService;
```

Comment:
can you try to add the icon service as public member in the constructor? this might work and save some lines and variables